### PR TITLE
docs/editions: Replace categories view with feature rich page

### PR DESCRIPTION
**Pull request by joebonrichie**
## Description

This replaces the categories view with a feature rich page for the editions/spins. It includes screenshots, a short description as well as direct links to the configuration and tips and tricks sub pages; all contained within a nice card view format.

A dependency is added on @docusaurus/plugin-ideal-image for automatic image resizing at deploy time.

### Submitter Checklist

- [ ] Squashed commits with `git rebase -i` (if needed)

### CODE REVIEW

File: src/components/editions/index.tsx [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+import clsx from 'clsx';
+import Translate from '@docusaurus/Translate';
+import Link from '@docusaurus/Link';
+import Image from '@theme/IdealImage';
+import Heading from '@theme/Heading';
+
+const Editions = [
+	{
+		name: 'Budgie',
+		image: require('@site/static/img/Budgie.jpg'),
```
Comment (JoshStrobl) [inline src/components/editions/index.tsx:20]: Should be doing an import instead of require.

File: src/components/editions/index.tsx [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+		urlTips: 'budgie/tips-and-tricks',
+		description: (
+			<Translate id="edition.budgie.description">
+			A feature-rich, luxurious desktop using the most modern technologies.
+			</Translate>
+		),
+	},
+	{
+		name: 'Plasma',
+		image: require('@site/static/img/Plasma.jpg'),
```
Comment (JoshStrobl) [inline src/components/editions/index.tsx:32]: Should be doing an import instead of require.

File: src/components/editions/index.tsx [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+		urlTips: 'plasma/tips-and-tricks',
+		description: (
+			<Translate id="edition.plasma.description">
+			A sophisticated desktop experience for the tinkerers. Simple by default, powerful when needed.
+			</Translate>
+		),
+	},
+	{
+		name: 'GNOME',
+		image: require('@site/static/img/GNOME.jpg'),
```
Comment (JoshStrobl) [inline src/components/editions/index.tsx:44]: Should be doing an import instead of require.

File: src/components/editions/index.tsx [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+		urlTips: 'gnome/tips-and-tricks',
+		description: (
+			<Translate id="edition.gnome.description">
+			A simple, streamlined desktop for more modern hardware.
+			</Translate>
+		),
+	},
+	{
+		name: 'MATE',
+		image: require('@site/static/img/MATE.jpg'),
```
Comment (JoshStrobl) [inline src/components/editions/index.tsx:56]: Should be doing an import instead of require.

File: src/components/editions/index.tsx [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+		<Link className="button button--secondary" to={urlConfig}>
+		Configuration
+		</Link>
+		<Link className="button button--secondary" to={urlTips}>
+		Tips & Tricks
+		</Link>
+		</div>
+		</div>
+		</div>
+		</div>
```
Comment (JoshStrobl) [inline src/components/editions/index.tsx:103]: Indentation needs fixing, should be using translatables here, don't use div use Box from MUI, prefer MUI Button and Button Group, etc.

File: src/components/editions/index.tsx [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+	);
+}
+
+export function EditionCardsRow(): JSX.Element {
+	return (
+		<div className="row">
```
Comment (JoshStrobl) [inline src/components/editions/index.tsx:109]: If we're doing a grid, this should be a GridItem and use Unstable_Grid from MUI. If we're doing a vertical list, use a Stack.

**Review by JoshStrobl — CHANGES_REQUESTED** [review]

File: src/components/editions/index.tsx [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+		urlTips: 'tips-and-tricks',
+		description: (
+			<Translate id="edition.mate.description">
+			A traditional desktop for advanced users and older hardware.
+			</Translate>
+		),
+	},
+];
+
+interface Props {
```
Comment (JoshStrobl) [inline src/components/editions/index.tsx:68]: Should not be interface, use `type` e.g. `type EditionCardProps = {`

File: src/components/editions/index.tsx [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+	name: string;
+	image: string;
+	url: string;
+	urlConfig: string;
+	urlTips: string;
+	description: JSX.Element;
+}
+
+function EditionCard({name, image, url, urlConfig, urlTips, description}: Props) {
```
Comment (JoshStrobl) [inline src/components/editions/index.tsx:77]: So not use function, do: `const EditionCard = ({ ... }: EditionCardProps) => {`

File: src/components/editions/index.tsx [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+		Tips & Tricks
+		</Link>
+		</div>
+		</div>
+		</div>
+		</div>
+	);
+}
+
+export function EditionCardsRow(): JSX.Element {
```
Comment (JoshStrobl) [inline src/components/editions/index.tsx:107]: Do not use function, use `const` arrow funcs. You also don't need to specify that it is a JSX.Element if you use the right elements.

**Comment by JoshStrobl** [comment]
Please be sure to use `yarn prettier` as well.

**Comment by joebonrichie** [comment]
Didn't end up using MUI buttons as I can't say I liked the look of them

**Comment by joebonrichie** [comment]
Use MUI Buttons

**Review by JoshStrobl — APPROVED** [review]
LGTM :rocket:

**Pull request merged**